### ISSUE TITLE
After struggling with openssl

### ISSUE THREAD

**Issue by yfery** [khyperia/weechat-discord#26]
It's not an issue but a feedback after struggling with openssl.

First, I'm using debian9 stable. 

For building `weechat-discord` we need openssl1.0 because `openssl-sys-extras v0.7.14` only support 1.0 version (1.1 support come with version 0.9 of `openssl-sys-extras`)

So for debian9 we need: ` sudo apt install pkg-config libssl1.0-dev libssl1.0.2 `

Be careful `libssl-dev` is 1.1 version, so not compatible yet.

It could be useful to add a subsection into readme about this particular point.

NB: pkg-config allows rust compiler (and others building tools) to find path information of libssl, without setting those path yourself via env vars (`OPENSSL_INCLUDE_DIR` and `OPENSSL_LIB_DIR`)

**Comment by khyperia** [comment]
Sorry about that - I thought I documented that, but I'm not seeing it.

> NB: pkg-config allows rust compiler (and others building tools) to find path information of libssl, without setting those path yourself via env vars (OPENSSL_INCLUDE_DIR and OPENSSL_LIB_DIR)

Unfortunately this doesn't work too well :( because openssl-1.0 is needed, and some distros are on 1.1, the story gets complicated. Some distros require the package openssl-1.0 (or libssl1.0, etc.), which using pkg-config doesn't work against, because it finds the openssl 1.1 .pc file and reports the include/link flags for that package.

The reason why I set those env vars manually is to override the default behavior to use openssl-1.1 and drop it to 1.0. It's a really sad compat story, but until discord-rs updates its dependency to openssl-sys-extras 0.9, it's required.
